### PR TITLE
[match] fix match nuke not deleting decrypted files

### PR DESCRIPTION
**Pull request by revolter**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

I noticed that running `fastlane match nuke` and cancelling it (answering `n` to "Do you really want to nuke everything listed above? (y/n)") causes the decrypted git repo to remain in the temp folder, which I personally think it's a security issue.

### Description

I added an `ensure` block that deletes it, similar to the `runner.rb` file.

### Testing Steps

Add
```ruby
gem 'fastlane', git: 'https://github.com/revolter/fastlane.git', branch: 'security/match-nuke-cancel-decrypted-files'
```
to your `Gemfile`, and run `bundle install` to apply the changes.

Then run `bundle exec fastlane match nuke development`, then press `n`, and see that a folder like `/var/folders/73/rlp8bl_x5lg1vbv1xbz159d40000gn/T/d20221021-22249-cq51af` still exists, which contains the decrypted certificates and profiles.

### CODE REVIEW

File: match/lib/match/commands_generator.rb [hunk 1]
```diff
@@ -155,7 +155,8 @@ def run
         FastlaneCore::CommanderGenerator.new.generate(Match::Options.available_options, command: c)
 
         c.action do |args, options|
-          Match::Migrate.new.migrate(args, options)
+          params = FastlaneCore::Configuration.create(Match::Options.available_options, options.__hash__)
+          Match::Migrate.new.migrate(params)
```
Comment (rogerluan) [inline match/lib/match/commands_generator.rb:159]: Why this change was needed? Was it to allow for direct calls of this command via Ruby?

**Review by rogerluan — APPROVED** [review]
LGTM! Great find!! 🙌

File: match/lib/match/commands_generator.rb (hunk 1 above)
Comment (revolter) [reply match/lib/match/commands_generator.rb:159]: To be able to call it from the tests:

https://github.com/fastlane/fastlane/pull/20776/files/868db12e6781cffc6347a31fee4e2ad84eb31167#diff-a3fda74b644561adcfae676cad5b8ca95f7ee0d7edea8ebcc54c359926e118c2R51

Otherwise, this:

https://github.com/fastlane/fastlane/pull/20776/files/868db12e6781cffc6347a31fee4e2ad84eb31167#diff-f79e3f0b905679ef7e333b8a455ee64c7b44ca4ecab8ca7cf487d479ef5f33dbL11

fails with an error stating that there is no `__hash__` method or something.

**Review by joshdholtz — APPROVED** [review]
Oh this is amazing! Thanks for finding and fixing this 😊 

Also... hello 👋 🥰

**Comment by revolter** [comment]
Hello 🤗 Thanks for merging!
